modules/images/webp-uploads/image-edit.php

```diff
@@ -187,7 +187,10 @@ function ( $metadata, $post_meta_id ) use ( $post_id, $file_path, $mime_type, $e
 					// Create a file with then new extension out of the targeted file.
 					$target_file_name     = preg_replace( "/\.$current_extension$/", ".$extension", $thumbnail_file );
 					$target_file_location = path_join( $original_directory, $target_file_name );
-					$result               = $editor->save( $target_file_location, $targeted_mime );
+
+					remove_filter( 'image_editor_output_format', 'webp_uploads_filter_image_editor_output_format', 10, 3 );
+					$result = $editor->save( $target_file_location, $targeted_mime );
+					add_filter( 'image_editor_output_format', 'webp_uploads_filter_image_editor_output_format', 10, 3 );
```

Comment:
Probably makes sense to do this here, but still see my comment above. I think we have another problem that is more critical and we should focus on.

If the output configuration for `image/jpeg` does only include a single MIME type (`image/webp`), I don't even think this code needs to be run at all. WordPress core handles a single MIME type perfectly fine, so all of this here seems irrelevant. It is needed for multiple MIME types, but when the output filter says to only use one output MIME type, almost nothing of the module is needed in fact (so we could add early returns to a bunch of entry filter callbacks).

Maybe I'm oversimplifying this, but again, I don't think we should _need_ to temporarily remove the filter here and elsewhere in this PR, the problems seem to be somewhere else. If our tests are failing without this, maybe it is because our tests assume there are multiple MIME types in the output for each MIME type? Maybe for those tests, we'll need to manually add a filter to ensure those tests always run with multiple MIME types (e.g. `'image/jpeg' => array( 'image/webp', 'image/jpeg' )`).

Reply:
This is added for the same reason as in the case above.

Reply:
Temporary filter removal LGTM, however note there are other flaws we need to still address elsewhere.